core/src/test/java/org/infinispan/tx/EntryWrappingInterceptorDoesNotBlockTest.java

```diff
@@ -0,0 +1,280 @@
+      for (int i = 0; i < keys.length; i++) {
+         MagicKey key = keys[i];
+         assertEquals("v" + i, cache(2).get(key));
+      }
+
+      crm0.stopBlocking();
+      crm1.stopBlocking();
+
+      newNode.get(10, TimeUnit.SECONDS);
+   }
+
+   private Object readWriteKey(MagicKey key, int index) {
+      FunctionalMap.ReadWriteMap<Object, Object> rwMap = ReadWriteMapImpl.create(FunctionalMapImpl.create(cache(0).getAdvancedCache()));
+      CompletableFuture cf = rwMap.eval(key, (Function<EntryView.ReadWriteEntryView<Object, Object>, Object> & Serializable) view -> {
+         assertFalse(view.find().isPresent());
+         view.set("v" + index);
+         return "r" + index;
+      });
+      return cf.join();
+   }
+
+   private Object readWriteMany(MagicKey key, int index) {
+      // make sure the other key is stable
+      MagicKey otherKey = new MagicKey("other", cache(0), cache(2));
+      FunctionalMap.ReadWriteMap<Object, Object> rwMap = ReadWriteMapImpl.create(FunctionalMapImpl.create(cache(0).getAdvancedCache()));
+      HashSet<MagicKey> keys = new HashSet<>(Arrays.asList(key, otherKey));
+      Traversable<Object> traversable = rwMap.evalMany(keys, (Function<EntryView.ReadWriteEntryView<Object, Object>, Object> & Serializable) view -> {
+         assertFalse(view.find().isPresent());
+         view.set("v" + index);
+         return "r" + index;
+      });
+      return traversable.findAny().orElseThrow(IllegalStateException::new);
+   }
+
+   private Object readWriteKeyValue(MagicKey key, int index) {
+      FunctionalMap.ReadWriteMap<Object, Object> rwMap = ReadWriteMapImpl.create(FunctionalMapImpl.create(cache(0).getAdvancedCache()));
+      CompletableFuture cfa = rwMap.eval(key, "v" + index, (BiFunction<Object, EntryView.ReadWriteEntryView<Object, Object>, Object> & Serializable) (value, view) -> {
+         assertFalse(view.find().isPresent());
+         view.set(value);
+         return "r" + index;
+      });
+      return cfa.join();
+   }
+
+   private Object readWriteManyEntries(MagicKey key, int index) {
+      // make sure the other key is stable
+      MagicKey otherKey = new MagicKey("other", cache(0), cache(2));
+      FunctionalMap.ReadWriteMap<Object, Object> rwMap = ReadWriteMapImpl.create(FunctionalMapImpl.create(cache(0).getAdvancedCache()));
+      HashMap<MagicKey, Object> map = new HashMap();
+      map.put(key, "v" + index);
+      map.put(otherKey, "something");
+      Traversable<Object> traversable = rwMap.evalMany(map, (BiFunction<Object, EntryView.ReadWriteEntryView<Object, Object>, Object> & Serializable) (value, view) -> {
+         assertFalse(view.find().isPresent());
+         view.set(value);
+         return "r" + index;
+      });
+      return traversable.findAny().orElseThrow(IllegalStateException::new);
+   }
+
+   private Object putDeltaWrite(MagicKey key, int index) {
+      // adding DELTA_WRITE is sufficient to force LoadType.OWNERS, we don't have to do actual delta
```

Comment:
I guess there's no class check because the parameter to a `DELTA_WRITE` put is a `DeltaAware` on the originator and a `Delta` on the remote nodes. Still, it looks weird :)